luigi/contrib/bigquery.py

```diff
@@ -507,40 +529,155 @@ def run(self):
         output = self.output()
         assert isinstance(output, BigQueryTarget), 'Output must be a BigQueryTarget, not %s' % (output)
 
+        project_id = output.table.project_id
+
         bq_client = output.client
+        gcs_client = GCSClient()
 
         source_uris = self.source_uris()
         assert all(x.startswith('gs://') for x in source_uris)
 
-        job = {
+        partial_table_ids = []
+        load_start_time = time.time()
+
+        def submit_load_job(uris, table_id):
+            job = {
+                'projectId': output.table.project_id,
+                'configuration': {
+                    'load': {
+                        'destinationTable': {
+                            'projectId': output.table.project_id,
+                            'datasetId': output.table.dataset_id,
+                            'tableId': table_id,
+                        },
+                        'encoding': self.encoding,
+                        'sourceFormat': self.source_format,
+                        'writeDisposition': self.write_disposition,
+                        'sourceUris': uris,
+                        'maxBadRecords': self.max_bad_records,
+                        'ignoreUnknownValues': self.ignore_unknown_values
+                    }
+                }
+            }
+
+            if self.source_format == SourceFormat.CSV:
+                job['configuration']['load']['fieldDelimiter'] = self.field_delimiter
+                job['configuration']['load']['skipLeadingRows'] = self.skip_leading_rows
+                job['configuration']['load']['allowJaggedRows'] = self.allow_jagged_rows
+                job['configuration']['load']['allowQuotedNewlines'] = self.allow_quoted_new_lines
+
+            if self.schema:
+                job['configuration']['load']['schema'] = {'fields': self.schema}
+
+            print "Job :" + str(job)
+
+            bq_client.run_job(output.table.project_id, job, dataset=output.table.dataset)
+            if output.enable_chunking:
+                partial_table_ids.append(output.table.project_id + "." + output.table.dataset_id + "." + table_id)
```

Comment:
Astonishing conditional and side effect. Should just `return` that string? Or there could be a separate function to build the partial table id altogether, as this line doesn't use anything that was computed in this function's body. There would be one `if` less then.